### PR TITLE
Allow for multiple applications to share a Redis DB

### PR DESCRIPTION
**Pull request by TALlama**
Implements #400

Adds the ability to set a `key_prefix` which will be prepended to all Redis keys. This allows multiple applications to keep their schedules in the same Redis database without clobbering each other.

This change does not affect the runtime of the RSpec suite, but allows for the use case mentioned. Doing the first two commits without the last one would allow for monkey patches, but if we could do the whole thing that would be great.

### CODE REVIEW

**Review by marcelolx — COMMENTED** [review]
Can we get some documentation about it added to the README.md?

**Comment by TALlama** [comment]
@marcelolx added.

**Comment by TALlama** [comment]
Also heads up that the link to [the commit message style guide](https://github.com/sidekiq-scheduler/guides/tree/master/style/commit-message#commit-message-style-guide) on [CONTRIBUTING.md](https://github.com/sidekiq-scheduler/sidekiq-scheduler/blob/master/CONTRIBUTING.md) is broken

**Comment by marcelolx** [comment]
Thanks @TALlama! I'll look into updating the contributing guide.

Added the PR to my list to review, as soon as I have time I'll take a final look

**Review by marcelolx — APPROVED** [review]
Thank you @TALlama! It looks good to me, there is just a conflict, once that is fixed I'll merge this PR (if you don't have time, don't mind, I can fix it)

**Comment by marcelolx** [comment]
Thanks @TALlama! I have to fix the CI to run the tests, but from what I remember the last time it did pass, so probably nothing is broken.

Thank you for your patience!

**Pull request merged**
